docs/checks/internal/validate/main.go

```diff
@@ -15,67 +15,266 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
+	"path"
+	"regexp"
 	"strings"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
 	"github.com/ossf/scorecard/v3/checks"
 	docs "github.com/ossf/scorecard/v3/docs/checks"
 )
 
 var (
 	allowedRisks     = map[string]bool{"Critical": true, "High": true, "Medium": true, "Low": true}
 	allowedRepoTypes = map[string]bool{"GitHub": true, "local": true}
+	supportedAPIs    = map[string][]string{
+		// InitRepo is supported for local reepos in general. However, in the context of checks,
+		// this is only used to look up remote data, e.g. in Fuzzinng check.
+		// So we only have "GitHub" supported.
+		"InitRepo":                   {"GitHub"},
+		"URI":                        {"GitHub", "local"},
+		"IsArchived":                 {"GitHub"},
+		"ListFiles":                  {"GitHub", "local"},
+		"GetFileContent":             {"GitHub", "local"},
+		"ListMergedPRs":              {"GitHub"},
+		"ListBranches":               {"GitHub"},
+		"GetDefaultBranch":           {"GitHub"},
+		"ListCommits":                {"GitHub"},
+		"ListReleases":               {"GitHub"},
+		"ListContributors":           {"GitHub"},
+		"ListSuccessfulWorkflowRuns": {"GitHub"},
+		"ListCheckRunsForRef":        {"GitHub"},
+		"ListStatuses":               {"GitHub"},
+		"Search":                     {"GitHub", "local"},
+		"Close":                      {"GitHub", "local"},
+	}
 )
 
+// Indentify the source file that declares each check.
+func listCheckFiles() (map[string]string, error) {
+	checkFiles := make(map[string]string)
+	// Use regex to determine the file that contains the entry point.
+	// We're looking for `const someVarName = "CheckName"`.
+	regex := regexp.MustCompile(`const\s+[^"]*=\s+"(.*)"`)
+	files, err := ioutil.ReadDir("checks/")
+	if err != nil {
+		return nil, fmt.Errorf("ioutil.ReadDir: %w", err)
+	}
+
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), ".go") || file.IsDir() {
+			continue
+		}
+
+		fullpath := path.Join("checks/", file.Name())
+		content, err := ioutil.ReadFile(fullpath)
+		if err != nil {
+			return nil, fmt.Errorf("ioutil.ReadFile: %s: %w", fullpath, err)
+		}
+
+		res := regex.FindStringSubmatch(string(content))
+		if len(res) != 2 {
+			continue
+		}
+
+		r := res[1]
+		if entry, exists := checkFiles[r]; exists {
+			//nolint:goerr113
```

Comment:
Can we avoid this? Not a blocker. Just wanted to call out.

Reply:
need a pass to remove all of them, right. Fortunately this is not part of an API we expose.